packages/esm-service-queues-app/src/types/index.ts

```diff
@@ -437,3 +421,125 @@ export interface WaitTime {
   metric: string;
   averageWaitTime: string;
 }
+
+export interface QueueTableColumn {
+  headerI18nKey: string;
+  cellComponent: (queueEntry: QueueEntry) => JSX.Element;
+}
+
+export interface QueueTableTabConfig {
+  columns: QueueTableColumn[];
+}
+
```

Comment:
@ibacher / @denniskigen / @brandones - I'm super surprised that the various interfaces we added below are not long-since added and available in esm-api.  Shouldn't we have common interfaces that represent our REST API that everyone shares?  We added the below here, but this really should be moved to the framework and all of the various esms can just import these common interfaces.  Is this omission intentional, or just never done?  Seems very surprising.

Reply:
Agree that these should exist in esm-api... a few things to think about:

* We will need to have a naming convention that supports interfaces for both the REST WS and FHIR representations (ie OMRSPatient vs FHIRPatient or something like that).
* We likely will need liberal use of "?" for optional fields to allow these interfaces to be used for full, default, and custom refs of an object.

Reply:
We intentionally take a slow and cautious approach to adding stuff to core. The right time to add it is when we start thinking it's crazy it wasn't added a long time ago. No regrets yet. The Queue stuff, idk, but the Person stuff, sure, seems like it's time.

Reply:
I haven't looking deeply into the code, but my thought would be if we think the the Queue REST Response provided the queue module is relatively stable, we should go ahead and create an interface for it that maps to the REST response.  This can live in the Queue module or the EMR API depending about our confidence in the stability.  Once we get into the the idea of "a mapped interface of how we want to deal with queue's on the client side" we should generally be more cautious.

Reply:
I think the real test here is not how _stable_ we think something is, but how _likely it is to be needed in multiple places_.

I can buy both maintaining types for core, stable types that are used in many places (things like `OpenmrsResource`, `OpenmrsSearchResult`, maybe `Concept`, `Provider`, and `Person`). Stuff that is app specific (queues and probably queue entries) probably is best defined in the queues app.

We _shouldn't_ need to come up with our own FHIR types... the `fhir` typescript namespace should already be available, e.g. `fhir.Patient`. Also, we need to be realistic about the surface area we can cover: most of the newly-added stuff isn't even documented in our [REST docs](https://rest.openmrs.org/#openmrs-rest-api) (and the REST module's Swagger docs are often not helpful).

Reply:
@ibacher  good to know that these have already been built by others for FHIR.